c2rust-transpile/src/translator/assembly.rs

```diff
@@ -472,10 +472,55 @@ fn asm_is_att_syntax(asm: &str) -> bool {
     }
 }
 
+/// If applicable, maps the index of an input operand to the
+/// output operand it is tied to.
+///
+/// Utilizes the fact that GNU inline assembly specifies all
+/// operands in a particular order of [output1, ... outputN]
+/// followed by [input1, ..., inputN] where the indices for all
+/// input operands are indexed relative to the size of the
+/// output operand sequence.
+fn remapped_idx(
+    idx_str: &str,
+    num_output_operands: usize,
+    tied_operands: &HashMap<(usize, bool), usize>,
+) -> String {
```

Comment:
I think this would compose better if it took `idx: usize` (rather than `&str`) and returned a `usize`, as this shrinks the domain and pushes parse errors back into the parser in `rewrite_asm` (which might want to return a `Result`). We really don't want malformed asm to cause the whole transpiler to fail if we can avoid it.